nixos/manpages: Explain nixos-rebuild -I option and how to build manpages (#205311)
NixOS/nixpkgs: nixos/doc/manual/contributing-to-this-manual.chapter.md

# Contributing to this manual {#chap-contributing}

> The [DocBook] and CommonMark sources of NixOS' manual are in the [nixos/doc/manual](https://github.com/NixOS/nixpkgs/tree/master/nixos/doc/manual) subdirectory of the [Nixpkgs](https://github.com/NixOS/nixpkgs) repository.

Reviewer: ```suggestion
The [DocBook] and CommonMark sources of the NixOS manual are in the [nixos/doc/manual](https://github.com/NixOS/nixpkgs/tree/master/nixos/doc/manual) subdirectory of the [Nixpkgs](https://github.com/NixOS/nixpkgs) repository.
```

```

If the build succeeds, the manual will be in `./result/share/doc/nixos/index.html`.

**Contributing to the man pages**

The man pages are written in [DocBook] which is XML.

To see what your edits look like:

```ShellSession
$ cd /path/to/nixpkgs
$ nix-build nixos/release.nix -A manpages.x86_64-linux
```

You can then read the man page you edited by running

```ShellSession
$ man --manpath=result/share/man nixos-rebuild # Replace nixos-rebuild with the command whose manual you edited
```

> If you're on a different architecture that's supported by nixOS (check nixos/release.nix) then replace `x86_64-linux` by the architecture.

Reviewer: ```suggestion
If you're on a different architecture that's supported by NixOS (check nixos/release.nix) then replace `x86_64-linux` with the architecture.
```

Author: I actually had to look this up and learned something new again: 

 - replace with = active voice
 - replace by = passive voice

Will modify

Reviewer: Yeah, go figure

Author: Done

```

If the build succeeds, the manual will be in `./result/share/doc/nixos/index.html`.

**Contributing to the man pages**

> The manpages are written in [DocBook] which is XML.

Reviewer: ```suggestion
The man pages are written in [DocBook] which is XML.
```

Author: Done

To see what your edits look like:

> ```ShellSession
> $ cd /path/to/nixpkgs
> $ nix-build nixos/release.nix -A manpages.x86_64-linux
> ```

Reviewer: ````suggestion
```
nix-build nixos/release.nix -A manpages.x86_64-linux
```
````

Author: This doesn't make it clear from where to execute the command. Someone coming from the `README.md` in the same folder will not know that they have to `cd` to the repo root in order to execute the command.

Additionally, this diverges from the format of the commands higher up in the file.

https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/contributing-to-this-manual.chapter.md?plain=1

Reviewer: Fair, let's keep the format unified.

You can then read the man page you edited by running

> ```ShellSession
> theCommand=nixos-rebuild  # Replace this with the command whose manual you edited
> man --manpath=result/share/man $theCommand
> ```

Reviewer: ````suggestion
```
man --manpath=result/share/man nixos-rebuild # replace with the command whose man page you edited
```
````

Author: This doesn't make it clear *which* command to replace `man`, `nixos-rebuild`, or the entire line. I'll `sed s/this/nixos-rebuild` though. That should make it more clear

Author: I made it one line and amended the comment.

Could you have another look @ncfavier ?

> If you're on a different architecture that's supported by nixOS (check nixos/release.nix)
>  then replace `x86_64-linux` by the architecture.

Reviewer: ```suggestion
If you're on a different architecture that's supported by NixOS (check nixos/release.nix) then replace `x86_64-linux` with the architecture.
```

Author: Done
